### PR TITLE
fix genomic interpretation ingestion

### PR DESCRIPTION
**Pull request by v-rocheleau**
Issue: The Interpretation/Diagnosis relationship (many-to-one) is incorrect. When ingesting Interpretations with Diagnoses that share the same Disease, GenomicInterpretations are associated to the wrong Diagnosis.

Solution: Change the Interpretation/Diagnosis relationship to a one-to-one

### CODE REVIEW

File: chord_metadata_service/chord/ingest/phenopackets.py [hunk 1]
```diff
@@ -236,14 +236,14 @@ def get_or_create_genomic_interpretation(gen_interp: dict) -> pm.GenomicInterpre
...
-    gen_obj, _ = pm.GenomicInterpretation.objects.get_or_create(
+    gen_obj, created = pm.GenomicInterpretation.objects.get_or_create(
         interpretation_status=gen_interp["interpretation_status"],
         gene_descriptor=gene_descriptor,
         variant_interpretation=variant_interpretation,
         extra_properties=gen_interp.get("extra_properties", {}),
     )
 
-    if related_obj:
+    if related_obj and created:
```
Comment (davidlougheed) [inline chord_metadata_service/chord/ingest/phenopackets.py]: why this change? I'm not sure what the original problem was, but if the object is re-used and the related object doesn't already have the object as an interpretration, shouldn't it still be added?

File: chord_metadata_service/chord/ingest/phenopackets.py [hunk 2]
```diff
@@ -262,26 +262,36 @@ def get_or_create_disease(disease) -> pm.Disease:
...
-        for gen_interp
-        in genomic_interpretations_data
-    ]
-    # disease = pm.Disease.objects.get_or_create(diagnosis["disease"])
-    diag_obj, _ = pm.Diagnosis.objects.get_or_create(
+def get_or_create_interpretation_diagnosis(interpretation: dict) -> pm.Diagnosis:
+    diagnosis = interpretation.get("diagnosis")
+
+    if not diagnosis:
+        return
```
Comment (davidlougheed) [inline chord_metadata_service/chord/ingest/phenopackets.py:280]: this should use a logger to log a warning/error and in the future generate an ingestion warning in the report, i think

Comment (davidlougheed) [reply chord_metadata_service/chord/ingest/phenopackets.py:280]: also this violates the return type hint, since the type hint says there will always be a returned Diagnosis

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: chord_metadata_service/chord/ingest/phenopackets.py (hunk 1 above)
Comment (v-rocheleau) [reply chord_metadata_service/chord/ingest/phenopackets.py]: Removed, I was testing stuff out and forgot to remove this check.

File: chord_metadata_service/chord/ingest/phenopackets.py (hunk 2 above)
Comment (v-rocheleau) [reply chord_metadata_service/chord/ingest/phenopackets.py:280]: An Interpretation may have no Diagnosis, so this shouldn't be logged as an error. The function creates a Diagnosis from an Interpretation if it contains one, otherwise it returns None. I corrected the return type.

**Review by davidlougheed — APPROVED** [review]
looks good to me - same comment about maybe prefixing the `related_type`

**Review by davidlougheed — COMMENTED** [review]
I like the logic for the extra props prefixes! maybe we should factor out `__` and the builder fn for prop names (i.e., `__` + name) into constants/a function but the logic looks solid!

**Review by davidlougheed — APPROVED** [review]
lgtm
